### ISSUE TITLE
Anonymous Struct in Function Parameter leads to C Error

### ISSUE THREAD

**Issue by timbasel** [vlang/v#15695]
**V version:** V 0.3.1 862d91e.1c63ce4
**OS:** Arch Linux 5.19.6

**What did you do?**
```v
module main

fn function(options struct { foo string = 'bar' }) {
	println(options.foo)
}

fn main() {
	function(struct { 
		foo: 'baz' 
	})
}
```

**What did you expect to see?**
A working program, printing `baz`
 
**What did you see instead?**

```
==================
/usr/lib64/crt1.o: error: Invalid relocation entry [15] '.rela.debug_info' @ 00000504
tcc: error: file 'crt1.o' not found
/tmp/v_1000/01.10568873222518203306.tmp.c:12065: error: identifier expected
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

**Comment by Delta456** [comment]
Should this syntax be valid?

**Comment by timbasel** [comment]
I would think if anonymous structs are supported inside of structs they should be supported everywhere else too.

**Comment by spytheman** [comment]
Fixed by bce1039 .
